game/src/client/3d/models/Knight/NPC.tsx

```diff
@@ -7,21 +7,61 @@ import { z } from "zod";
 import { useFrame } from "@react-three/fiber";
 import { Group, Vector3 } from "three";
 import { proxy, useSnapshot } from "valtio";
-import { npcPosition, playerPosition } from "../../../state/positions";
+import { GameEntityProps, NPCPositions, playerPosition } from "../../../state/positions";
 import { SpeechBubble } from "../../components/SpeechBubble";
 import Robot from "./Robot";
-
+import Alien from "./Alien";
 export const npcPlayerVisual = proxy({
   rollCooldown: false,
   rolling: false,
   moving: false,
   running: false,
 });
 
+
+function GameEntity({ type, localPlayerState }:GameEntityProps) {
+  const npcRef = useRef(null);
+
+  // Render the appropriate entity based on the `type` prop
+  let EntityComponent;
+  switch (type) {
+    case 'robot':
+      EntityComponent = (
+        <Robot
+          ref={npcRef}
+          lastAttack={0}
+          lastDamaged={0}
+          moving={localPlayerState.moving}
+          recharging={false}
+          running={localPlayerState.rolling}
+        />
+      );
+      break;
+    case 'alien':
+      EntityComponent = (
+        <Alien
+          ref={npcRef}
+          lastAttack={0}
+          lastDamaged={0}
+          moving={localPlayerState.moving}
+          recharging={false}
+          running={localPlayerState.rolling}
+        />
+      );
+      break;
+    default:
+      EntityComponent = <div>Unknown entity type</div>;
```

Comment:
Instead of text, just throw an error here. React-three-fiber is not capable of rendering text in the middle of canvas.